odoo/models.py

```diff
@@ -6205,7 +6205,20 @@ def filtered_domain(self, domain):
                         model = model[fname]
 
                 if comparator in ('like', 'ilike', '=like', '=ilike', 'not ilike', 'not like'):
-                    value_esc = value.replace('_', '?').replace('%', '*').replace('[', '?')
+                    if comparator.endswith('ilike'):
+                        # ilike uses unaccent and lower-case comparison
+                        def unaccent_real(x):
+                            return self.pool.unaccent_python(x.lower()) if x else ''
+
+                        unaccent = unaccent_real
+                    else:
+                        def str_or_empty(x):
+                            return x or ''
+
+                        unaccent = str_or_empty
```

Comment:
Sry, but I still don't understand this indirection, why not just define def unaccent(x), directly ?

```suggestion
                        def unaccent(x):
                            return self.pool.unaccent_python(x.lower()) if x else ''
                    else:
                        def unaccent(x):
                            return x or ''
```
Or I miss something and I need a coffee ?

Reply:
I've updated to your code. I was just explicitely naming functions (sometimes I find it clearer).